packages/flutter/lib/src/painting/_network_image_web.dart

```diff
@@ -80,18 +82,24 @@ class NetworkImage
     NetworkImage key,
     image_provider.DecoderCallback decode,
     StreamController<ImageChunkEvent> chunkEvents,
-  ) {
+  ) async {
     assert(key == this);
 
     final Uri resolved = Uri.base.resolve(key.url);
-    // This API only exists in the web engine implementation and is not
-    // contained in the analyzer summary for Flutter.
-    return ui.webOnlyInstantiateImageCodecFromUrl(// ignore: undefined_function, avoid_dynamic_calls
-      resolved,
-      chunkCallback: (int bytes, int total) {
-        chunkEvents.add(ImageChunkEvent(cumulativeBytesLoaded: bytes, expectedTotalBytes: total));
-      },
-    ) as Future<ui.Codec>;
+
+    final HttpRequest response = await HttpRequest.request(key.url, method: 'GET', requestHeaders: key.headers, responseType: 'arraybuffer');
```

Comment:
In order for this code to be testable, instead of calling `HttpRequest.request` directly we should call an abstraction that can be mocked out in a test. For example, the engine uses an overridable factory function: https://github.com/flutter/engine/blob/9d7bd5a6bb6ed4614c9011f373b310bb24e7ddaf/lib/web_ui/lib/src/engine/canvaskit/image.dart#L36

If the framework doesn't already have an abstraction, we might want to create one.